### PR TITLE
Disable registration in QA_MODE

### PR DESCRIPTION
**Pull request by juadk**
Now that we have salt-minion included in public cloud images, registration is not needed anymore for HANA/cluster nodes. And we have to make sure to not install packages outside of images.
This change does not concern iscsi/monitoring servers.

Tests:
- AWS with and without QA_MODE: OK
- GCP with and without QA_MODE: OK
- Azure with and without QA_MODE: OK

I didn't test in libvirt but I think it is not impacted by this change, @arbulu89 @MalloZup WDYT?

### CODE REVIEW

**Comment by arbulu89** [comment]
Hi @juadk ,
My first question is. Are you sure we need this change? If salt is already installed it wouldn't actually apply this script, unless the `salt` binaries are located in other path. Maybe this is the case...
To improve that maybe we could just run some other command like `which salt-call` or similar to detect if salt is installed or not. 
What do you think @juadk ?

Second, what about Azure? Working as well?

**Comment by ldevulder** [comment]
> My first question is. Are you sure we need this change?

We mainly **NEED** to ensure that we **DO NOT** install anything on alpha/beta/RC version during Build Validation. This is why adding a test on `qa_mode` could be good.

But +1 for the `which`.

**Comment by arbulu89** [comment]
@juadk @ldevulder 
I think that my comment was misunderstood, so I will rewrite it.

I totally understand the goal of the PR, to avoid registration if you are running QA, this is clear.

My comment comes more about the implementation itself. The thing that the current images bring `salt` already embedded shouldn't impact the current implementation. The script checks if `salt` is installed, and if not, it registers the system and install it. So, if the image comes with `salt` as the latest version, the script won't be executed, so we don't really need to add this new `if` condition with the `QA_MODE` thing.

Actually, now, if you want to use some older version the new changes would make the deployment fail.
Summary: I wouldn't do this change. In any case, I would just update the line:
```
elif [[ ! -e /usr/bin/salt-minion ]]; then
```
to use the output of `which salt-call` (salt installed or not).

I don't know if I'm missing some point that requires this new change...

**Comment by ldevulder** [comment]
> Actually, now, if you want to use some older version the new changes would make the deployment fail.

Could be but we never did this, as Build Validation is only for new version. And the bad thing I can see is the case were a new version that we have to validate has no salt-minion installed, this will be a bug but in that specific case salt will be installed and we have the chance to not see this (as all this TF tests will be done through openQA shortly).

So for me it's better to add the test on `QA_MODE` to ensure. @juadk WDYT?

**Comment by juadk** [comment]
> So for me it's better to add the test on `QA_MODE` to ensure. @juadk WDYT?
Yes I agree with you, we need to know if salt-minion is removed from images. And as you mentioned, we will not do QA with older version.

@arbulu89 Maybe you didn't pay attention but I have already changed `/usr/bin/salt-minion` test with the `which` command.

File: salt/salt_provisioner_script.tpl [hunk 1]
```diff
@@ -2,9 +2,12 @@
 
 mv /tmp/salt /root || true
 
-# Check is qa_mode is enable
+# Check is qa_mode is enabled
```
Comment (arbulu89) [inline salt/salt_provisioner_script.tpl]: It should be `check **if** qa_mode is enabled`

Comment (juadk) [reply salt/salt_provisioner_script.tpl]: Thx

**Pull request merged**